lib/puppet/pops/types/type_calculator.rb

```diff
@@ -644,6 +644,11 @@ def infer_Symbol(o)
   end
 
   # @api private
+  def infer_Sensitive(o)
+    PSensitiveType.new(infer_set(o.unwrap))
```

Comment:
I know I told you to do this :-). With the generalize in the constructor, the rich information produced by `infer_set` will be erased. We could do a small optimization here, and use `infer` (as it generalizes).
OTOH, we then have the decision that the type generalizes in two places, and we cannot trust all types given to PSensitiveType.new to be generalized so it must be done there anyway.
Suggest keeping it as it is now. Just wanted to make you aware of this.